lib/core/utils/parse-sameorigin-stylesheets.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Parse non cross-origin stylesheets
+ *
+ * @param {Object} sheet CSSStylesheet object
+ * @param {Object} options options object from `axe.utils.parseStylesheets`
+ * @param {Array<Number>} priority sheet priority
+ * @param {Array<String>} importedUrls urls of already imported stylesheets
+ * @param {Boolean} isCrossOrigin boolean denoting if a stylesheet is `cross-origin`
+ * @returns {Promise}
+ */
+axe.utils.parseSameOriginStylesheets = function parseSameOriginStylesheets(
+	sheet,
+	options,
+	priority,
+	importedUrls,
+	isCrossOrigin = false
+) {
+	const rules = Array.from(sheet.cssRules);
+
+	if (!rules) {
+		return Promise.resolve();
+	}
+
+	/**
+	 * reference -> https://developer.mozilla.org/en-US/docs/Web/API/CSSRule#Type_constants
+	 */
+	const cssImportRules = rules.filter(r => r.type === 3); // type === 3 -> CSSRule.IMPORT_RULE
+
+	/**
+	 * when no `@import` rules in given sheet -> resolve the current `sheet` & exit
+	 */
+	if (!cssImportRules.length) {
+		// exit
+		return Promise.resolve({
+			isCrossOrigin,
+			priority,
+			root: options.rootNode,
+			shadowId: options.shadowId,
+			sheet
+		});
+	}
+
+	/**
+	 * filter rules that are not already fetched
+	 */
+	const cssImportUrlsNotAlreadyImported = cssImportRules
+		// ensure rule has a href
+		.filter(rule => rule.href)
+		// extract href from object
+		.map(rule => rule.href)
+		// only href that are not already imported
+		.filter(url => !importedUrls.includes(url));
+
+	/**
+	 * iterate `@import` rules and fetch styles
+	 */
+	const promises = cssImportUrlsNotAlreadyImported.map(
+		(importUrl, cssRuleIndex) => {
+			const newPriority = [...priority, cssRuleIndex];
+			const isCrossOriginRequest = /^https?:\/\/|^\/\//i.test(importUrl);
+
+			return axe.utils.parseCrossOriginStylesheets(
```

Comment:
Why is a same origin stylesheet going through the cross origin parse function? If the function's main purpose is to load a cross origin stylesheet or a same origin or cross origin import, we should probably rename the rule to reflect that it loads stylesheet urls (same or cross origin) and doesn't parse them.

Reply:
Given 

`@import "iam/a/relative/style.css"`
`@import "https://iam.io/external.css"`

The first one needs to be interpreted as `relative` with `isCrossOrigin` flag as `false`, and the other one as `true`. Have to use `axios` to fetch a relative or absolute resource & interpret it accordingly.